content/3-lifecycle/2-on-unmount/angular/time.component.ts

```diff
@@ -1,20 +1,10 @@
-import { Component, OnDestroy } from '@angular/core';
+import { Component } from '@angular/core';
+import { interval, map } from 'rxjs';
 
 @Component({
 	selector: 'app-time',
-	template: '<p>Current time: {{ time }}</p>',
+	template: '<p>Current time: {{ time$ | async }}</p>',
 })
-export class TimeComponent implements OnDestroy {
-	time: string = new Date().toLocaleTimeString();
-	timer: number;
-
-	constructor() {
-		this.timer = window.setInterval(() => {
-			this.time = new Date().toLocaleTimeString();
-		}, 1000);
-	}
-
-	ngOnDestroy(): void {
-		window.clearInterval(this.timer);
-	}
+export class TimeComponent {
+	time$ = interval(1000).pipe(map(() => new Date().toLocaleTimeString()));
```

Comment:
You want to shrink the snippet as little as possible but that's not the purpose here.
If `ngOnDestroy` is not used, the `onUnmount` section don't have sense anymore when comparing to other frameworks.
Here we don't actually care if the timer is optimized or not.

Reply:
You're correct, I will change this back to using ngOnDestroy

Reply:
I rolled this back.